src/main/java/io/cdap/plugin/gcp/gcs/source/GCSSource.java

```diff
@@ -119,16 +121,24 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     }
 
     Storage storage = GCPUtils.getStorage(config.connection.getProject(), credentials);
+    Bucket bucket = storage.get(bucketName);
+    if (Objects.isNull(bucket)) {
+      String errorReason = String.format("Unable to access GCS bucket '%s'.",
+              bucketName);
```

Comment:
indentation seems to be off here, please use `Google Configuration Checker Style`.